conduct/wallets/base.py

```diff
@@ -1,30 +1,75 @@
+import requests
+
 from abc import ABC, abstractmethod
-from typing import Optional
+from http import HTTPStatus
+from requests import Response
+
+from conduct.types import Invoice, MilliSatoshi
 
 
 class Wallet(ABC):
     """Abstract class."""
 
     @abstractmethod
-    async def get_info(self):
+    def get_info(self):
         pass
 
     @abstractmethod
-    async def get_balance(self):
+    def get_balance(self) -> MilliSatoshi:
         pass
 
     @abstractmethod
-    async def create_invoice(self, *, amount: Optional[int], memo: Optional[str]):
+    def create_invoice(
+        self, *, amount: int, description: str = "", expiry: int = 3600
+    ) -> Invoice:
         pass
 
     @abstractmethod
-    async def pay_invoice(self, *, payment_request: str):
+    def pay_invoice(self, *, payment_request: str):
```

Comment:
If there is a limit we could apply it at library level: read the payment_request ourselves and stop the payment if the maximum is reached. We don't need it implemented by the custodial services...

Reply:
Yeah, but then you have to know the max possible fee in advance. The request only tells you the amount that the sender shall receive, not how much will be sent. So you are back at guessing/modeling worst case scenarios (you might not even know the maximum fee some custodial service would make you pay).

Reply:
Yes, we can control the amount, but not the fees.

Reply:
Maybe an `AdvancedFeeMgmgt` mixin would work? Because this API will only allow for a super shitty UX of having x sats locked up forever because LNbits has to defend itself from fee attacks with one arm bound to its back :cry: with the proposed API this will go down to a few msat at least and eventually native backends could implement a sendall functionality by driving the route selection/payment themselves.